### PR TITLE
Adding user assigned managed identity support

### PR DESCRIPTION
**Pull request by serkantkaraca**
Adding support to define user assigned managed identity client id through connection string and client API.

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/TokenProvider.cs [hunk 1]
```diff
@@ -91,11 +91,20 @@ public static TokenProvider CreateSharedAccessSignatureTokenProvider(string keyN
...
-        /// <summary>Creates Azure Managed Identity token provider.</summary>
+        /// <summary>Creates system assigned Azure Managed Identity token provider.</summary>
         /// <returns>The <see cref="TokenProvider" /> for returning Json web token.</returns>
         public static TokenProvider CreateManagedIdentityTokenProvider()
         {
-            return new ManagedIdentityTokenProvider();
+            return CreateManagedIdentityTokenProvider(null);
+        }
+
+
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/TokenProvider.cs]: nit:  Extra blank

**Review by jsquire — APPROVED** [review]
With my limited knowledge of the `AppAuthentication` library, the changes look good to me.   I did notice that there are no tests for the additional scenarios; not sure if that was intended.

**Comment by jsquire** [comment]
@serkantkaraca:  Is this PR still pending, or should we close it out as stale?

**Comment by serkantkaraca** [comment]
> @serkantkaraca: Is this PR still pending, or should we close it out as stale?

I think we don't need this any more after custom token provider support is added. I am closing it now.

**Pull request closed without merging**